keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -516,12 +518,9 @@ def get_config(self):
         return {
             "num_classes": self.num_classes,
             "bounding_box_format": self.bounding_box_format,
-            # TODO(bischof): actually serialize the backbone
             "backbone": self.backbone,
-            "anchor_generator": self.anchor_generator,
```

Comment:
Won't removing this break serialization when a custom anchor_generator is used?

Reply:
Yes, this is a problem. I am not so sure about the logics between `anchor_generator`, `label_encoder`, `prediction_decoder`. When `get_config()` is called, they all exist. However, they cannot be all passed to `__init__()`. What would be a good way to judge which one to omit here?

Reply:
Okay actually I think what you've done is exactly right -- since the encoder/decoder are dependent on this, just serializing the encoder/decoder should be good.

This makes me wonder though -- @LukeWood why can't we pass a custom `anchor_generator` in addition to a custom `prediction_decoder`? It seems like those are now decoupled, so I wonder if this is an artifact of how this used to be structured?

Reply:
> Yes, this is a problem. I am not so sure about the logics between anchor_generator, label_encoder, prediction_decoder. When get_config() is called, they all exist. However, they cannot be all passed to __init__(). What would be a good way to judge which one to omit here?

We definitely should not  prevent the use of custom anchor generators.  One thing we could do is exclude anchor generators, and encourage only the use of `save_weights()` and `load_weights()`?  I'm not sure the right approach, but anchor generators and weights are tightly coupled unfortunately.

Reply:
@LukeWood -- since the label encoder contains the anchor generator, by serializing the label encoder aren't we also effectively serializing the anchor generator?

I feel like I'm missing something. Sorry if it's a stupid question!